code/script.js

```diff
@@ -1,325 +1,452 @@
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  // if the "make a guess" option (it's the first one so it has index 0) is selected, pressing "find out" should not do anything
+  if (questions.options[questions.selectedIndex].index === 0) {
+    return;
+  }
+
+  const { category, value } = currentQuestion;
+  numberOfGuesses++; // update the number of guesses
+  updateGuessCounter(); // and make sure to update the guess counter showing on the page also :)
 
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  if (category === "hair") {
+    if (value === secret.hair) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
+  } else if (category === "eyes") {
+    if (value === secret.eyes) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
+  } else if (category === "accessories" || category === "other") {
+    if (secret.accessories.includes(value) || secret.other.includes(value)) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
   }
-}
+};
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
   // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  if (category === "hair") {
     if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+      addToLog(`The person has ${value} ${category}!`);
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] === value
+      );
     } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+      addToLog(`The person doesn't have ${value} ${category}!`);
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] !== value
+      );
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === "eyes") {
+    if (keep) {
+      addToLog(`The person has ${value} ${category}!`);
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] === value
+      );
+    } else {
+      addToLog(`The person doesn't have ${value} ${category}!`);
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] !== value
+      );
+    }
+  } else if (category === "accessories") {
+    if (keep) {
+      addToLog(`The person wears ${value == "hat" ? "a" : ""} ${value}!`);
```

Comment:
When the player guesses on glasses it returns "The person does not wear false glasses". 
You could fix this by changing the option value in the CSS to "a hat" 
and then also change the key value in the array CHARACTERS to "a hat" instead of adding 
the a in the function which, in this case, returns false for the accessories that don't 
have an a in the array, such as glasses.